File: drills/elixir/harnesses/nc119_reconstruct_itinerary.exs
[
  {"find_itinerary(MUC/LHR/SFO/SJC chain)", inspect(["JFK", "MUC", "LHR", "SFO", "SJC"]),
   inspect(
     Solution.find_itinerary([
       ["MUC", "LHR"],
       ["JFK", "MUC"],
       ["SFO", "SJC"],
       ["LHR", "SFO"]
     ])
   )},
  {"find_itinerary(two ways out of JFK -- smallest first)",
   inspect(["JFK", "ATL", "JFK", "SFO", "ATL", "SFO"]),
   inspect(
     Solution.find_itinerary([
       ["JFK", "SFO"],
       ["JFK", "ATL"],
       ["SFO", "ATL"],
       ["ATL", "JFK"],
       ["ATL", "SFO"]
     ])
   )},
  {"find_itinerary(KUL is a dead end, so it must come last)",
   inspect(["JFK", "NRT", "JFK", "KUL"]),
   inspect(
     Solution.find_itinerary([["JFK", "KUL"], ["JFK", "NRT"], ["NRT", "JFK"]])
   )},
  {"find_itinerary([])", inspect(["JFK"]), inspect(Solution.find_itinerary([]))}
]

File: drills/ts/harnesses/nc119_reconstruct_itinerary.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.findItinerary !== "function") throw new Error("__signature_mismatch__");
  return [
    ["findItinerary(MUC/LHR/SFO/SJC chain)", show(["JFK", "MUC", "LHR", "SFO", "SJC"]), show(solution.findItinerary([["MUC", "LHR"], ["JFK", "MUC"], ["SFO", "SJC"], ["LHR", "SFO"]]))],
    ["findItinerary(two ways out of JFK -- smallest first)", show(["JFK", "ATL", "JFK", "SFO", "ATL", "SFO"]), show(solution.findItinerary([["JFK", "SFO"], ["JFK", "ATL"], ["SFO", "ATL"], ["ATL", "JFK"], ["ATL", "SFO"]]))],
    ["findItinerary(KUL is a dead end, so it must come last)", show(["JFK", "NRT", "JFK", "KUL"]), show(solution.findItinerary([["JFK", "KUL"], ["JFK", "NRT"], ["NRT", "JFK"]]))],
    ["findItinerary([])", show(["JFK"]), show(solution.findItinerary([]))],
  ];
}
